stablehlo/dialect/StablehloOps.cpp

```diff
@@ -2023,6 +2023,22 @@ LogicalResult TriangularSolveOp::inferReturnTypeComponents(
                                      inferredReturnShapes);
 }
 
+mlir::Speculation::Speculatability TriangularSolveOp::getSpeculatability() {
```

Comment:
>> Diagonal could be non-unit at runtime, leading to undefined behavior.
This seems odd to me. IOW, it would mean that the flag always has to be true for the op to be meaningful.
What the spec says is: if the flag is true, then the implementation has to assume that the diagonal elements are 1. Failing to assume that leads to undefined behavior. 

IIUC the following check says is meant for: if the flag is true, and if  implementation  fails to assume than the diagonal is 1, then there could be undefined behavior.
>>   `if (getUnitDiagonal()) return mlir::Speculation::NotSpeculatable;`

IMO, the above check is fine.

Reply:
We're on the same page. The comment only applies when `getUnitDiagonal` is `true`. I rewrote the comment to make that more obvious, thanks!